dotnet/src/Extensions/Planning.SequentialPlanner/SequentialPlanner.cs

```diff
@@ -57,16 +57,15 @@ public async Task<Plan> CreatePlanAsync(string goal, CancellationToken cancellat
             throw new SKException("The goal specified is empty");
         }
 
-        string relevantFunctionsManual = await this._context.GetFunctionsManualAsync(goal, this.Config, cancellationToken).ConfigureAwait(false);
-        this._context.Variables.Set("available_functions", relevantFunctionsManual);
+        string relevantFunctionsManual = await this._kernel.CreateNewContext().GetFunctionsManualAsync(goal, this.Config, cancellationToken).ConfigureAwait(false);
 
-        this._context.Variables.Update(goal);
+        ContextVariables vars = new(goal) { ["available_functions"] = relevantFunctionsManual };
```

Comment:
👀

Reply:
Not sure. I'll check the rules.

Reply:
We should probably use a `const` for this key to make this less error-prone.